standalone/src/main/java/filodb/standalone/SimpleProfiler.java

```diff
@@ -0,0 +1,468 @@
+package filodb.standalone;
+
+import java.io.FileOutputStream;
+import java.io.InterruptedIOException;
+import java.io.IOException;
+import java.io.OutputStream;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.ThreadInfo;
+import java.lang.management.ThreadMXBean;
+
+import java.lang.reflect.Method;
+
+import java.time.Instant;
+
+import java.nio.charset.StandardCharsets;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Simple profiler which samples threads and periodically logs a report to a file. When the
+ * process is cleanly shutdown, the profiler stops and reports what it has immediately. This
+ * makes it possible to use a very long report interval without data loss.
+ */
+public class SimpleProfiler {
+    /**
+     * Launches a global profiler, based on system properties.
+     *
+     * -Dfilodb.SimpleProfiler.sampleRateMillis=<n>
+     * -Dfilodb.SimpleProfiler.reportIntervalSeconds=<n>
+     * -Dfilodb.SimpleProfiler.topCount=<n>
+     * -Dfilodb.SimpleProfiler.outFile=<filename>
+     *
+     * In order for profiling to be enabled, all of the above properies must be set.
+     *
+     * @throws IOException if file cannot be opened
+     */
+    public static boolean launch() throws IOException {
+        String prefix = "filodb.SimpleProfiler.";
+
+        int sampleRateMillis = getIntProperty(prefix, "sampleRateMillis");
```

Comment:
I suggest we use typesafe config instead of system properties here.
This way, we can enable this for specific environments.
Modifying command line arguments per environment may not work for everyone.

Reply:
What do you suggest specifically? I don't want this profiler to turn into something much bigger, since it's just for debugging. I wouldn't expect anyone to run this in production.

Reply:
You could move the launch code into the class `FilodbCluster` and drive the start using Typesafe config that can be loaded into `FilodbSettings`.

Agree - this is not for prod. It could be enabled during load testing only.